types/p2p.yaml

```diff
@@ -0,0 +1,67 @@
+  NetworkIdentity:
+    type: object
+    properties:
+      peer_id:
+        $ref: './p2p.yaml#/PeerId'
+      enr:
+        $ref: './p2p.yaml#/ENR'
+      p2p_addresses:
+        type: array
+        items:
+          allOf:
+            - $ref: './p2p.yaml#/Multiaddrs'
+            - description: "Node's addresses on which eth2 rpc requests are served. [Read more](https://docs.libp2p.io/reference/glossary/#multiaddr)"
+      discovery_addresses:
+        type: array
+        items:
+          allOf:
+            - $ref: './p2p.yaml#/Multiaddrs'
+            - description: "Node's addresses on which is listening for discv5 requests. [Read more](https://docs.libp2p.io/reference/glossary/#multiaddr)"
+      metadata:
+        $ref: './p2p.yaml#/MetaData'
+
+  MetaData:
+    type: object
+    description: "Based on eth2 [Metadata object](https://github.com/ethereum/eth2.0-specs/blob/v0.12.0/specs/phase0/p2p-interface.md#metadata)"
+    properties:
+      seq_number:
+        allOf:
+          - description: "Uint64 starting at 0 used to version the node's metadata. If any other field in the local MetaData changes, the node MUST increment seq_number by 1."
+          - $ref: './primitive.yaml#/schemas/Uint64'
+      attnets:
+        allOf:
+          - description: "Bitvector representing the node's persistent attestation subnet subscriptions."
+          - $ref: "./primitive.yaml#/schemas/Hex"
+
+  Peer:
+    type: object
+    properties:
+      peer_id:
+        $ref: "./p2p.yaml#/PeerId"
+      enr:
+        $ref: "./p2p.yaml#/ENR"
+      address:
+        $ref: "./p2p.yaml#/Multiaddrs"
+      state:
+        type: string
+        enum: ["disconnected", "connecting", "connected", "disconnecting"]
+      direction:
+        type: string
+        enum: ["inbound", "outbound"]
+
+
+
+  PeerId:
+    type: string
+    description: "Cryptographic hash of a peer’s public key. [Read more](https://docs.libp2p.io/concepts/peer-id/)"
+    example: "QmYyQSo1c1Ym7orWxLYvCrM2EmxFTANf8wXmmE7DWjhx5N"
+
+  ENR:
+    type: string
+    description: "Ethereum node record. [Read more](https://eips.ethereum.org/EIPS/eip-778)"
+    example: "enr:-IS4QHCYrYZbAKWCBRlAy5zzaDZXJBGkcnh4MHcBFZntXNFrdvJjX04jRzjzCBOonrkTfj499SZuOh8R33Ls8RRcy5wBgmlkgnY0gmlwhH8AAAGJc2VjcDI1NmsxoQPKY0yuDUmstAHYpMa2_oxVtw0RW_QAdpzBQA8yWM0xOIN1ZHCCdl8"
+
+  Multiaddrs:
```

Comment:
I think it would be clearer to name this `Multiaddr` (no "s") to remove any ambiguity that it's singular rather than plural